### PR TITLE
Fixed warning in fourigui.py code

### LINKED ISSUE #22: Resolve Pytest 25 warnings
25 warnings produced 
https://github.com/diffpy/diffpy.fourigui/actions/runs/11220139070/job/31187644168

```
tests/integration_test.py: 7 warnings
tests/unit_test.py: 3 warnings
  /usr/share/miniconda/envs/test/lib/python3.12/site-packages/numpy/lib/_type_check_impl.py:468: DeprecationWarning: __array__ implementation doesn't accept a copy keyword, so passing copy=False failed. __array__ must implement 'dtype' and 'copy' keyword arguments.
    x = _nx.array(x, subok=True, copy=copy)
  /usr/share/miniconda/envs/test/lib/python3.12/site-packages/numpy/fft/_pocketfft.py:892: DeprecationWarning: `axes` should not be `None` if `s` is not `None` (Deprecated in NumPy 2.0). In a future version of NumPy, this will raise an error and `s[i]` will correspond to the size along the transformed axis specified by `axes[i]`. To retain current behaviour, pass a sequence [0, ..., k-1] to `axes` for an array of dimension k.
    return _raw_fftnd(a, s, axes, fft, norm, out=out)

tests/integration_test.py::TestGui::test_load_cube_testdataset1
tests/integration_test.py::TestGui::test_load_cube_testdataset2
tests/integration_test.py::TestGui::test_load_cube_testdataset3
tests/unit_test.py::TestGui::test_load_cube_nothing_loaded
tests/unit_test.py::TestGui::test_load_cube_something_loaded
  /usr/share/miniconda/envs/test/lib/python3.12/site-packages/diffpy/fourigui/fourigui.py:266: DeprecationWarning: __array__ implementation doesn't accept a copy keyword, so passing copy=False failed. __array__ must implement 'dtype' and 'copy' keyword arguments.
...

======================= 18 passed, 25 warnings in 1.92s ========================
```

### PR DESCRIPTION
**Pull request by Fil158**
Closes #22 Warning fixed in fourigui.py adding line 397 `axes = list(range(len(size)))` and modifying line 399 `fftholder = np.fft.fftn(fftholder, s=size, axes=axes, norm="ortho")`

### CODE REVIEW

File: news/warning.rst [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+* <news item>
+
+**Changed:**
+
+* <news item>
+
+**Deprecated:**
+
+* Deprecation warning fixed for axes in np.fft in fourigui.py
```
Comment (bobleesj) [inline news/warning.rst:11]: @Fil158 

Typically depreciated refers to classes and functions no longer support.  So you can put it under "Fixed" - please refer to the "Lessons from ..." GitLab guide for writing a news file.

**Comment by bobleesj** [comment]
@Fil158 The original issue mentioned here was created to ensure we re-visit to fix these warnings. But, what exact warnings are you fixing and how? Even if it's a simple fix, we will probably forget things later and also it will be useful for future maintainers.

**Comment by bobleesj** [comment]
> Warning fixed in fourigui.py adding line 397 `axes = list(range(len(size)))` and modifying line 399 `fftholder = np.fft.fftn(fftholder, s=size, axes=axes, norm="ortho")`

Thanks for this. Reviewers all can see what were changed. This information has no further value...

File: src/diffpy/fourigui/fourigui.py [hunk 2]
```diff
@@ -394,8 +394,9 @@ def fft(self):
         def perform_fft(fftholder):
             fftholder = np.nan_to_num(fftholder)
             size = list(fftholder.shape)
+            axes = list(range(len(size)))
```
Comment (sbillinge) [inline src/diffpy/fourigui/fourigui.py:397]: this is not very clear.  In the comments, let's have a brief discussion of what is deprecated in numpy2 and how we are fixing it.  Not too much, but presumably @Fil158 you googled something and found something out so it is helpful to share what you found out and a few words about how it affects us here and so your suggested fix.  It builds more confidence that we understand what we are doing.

Also, please make your PR title more informative. Which warning does this PR fix?  Not too long, but just enough.

I am not sure but there is almost certainly a more readable fix than `list(range(len(size)))`

**Review by sbillinge — COMMENTED** [review]
nice work....great to see.....   A bit more discussion needed.  Please see comments/questions.

File: news/warning.rst (hunk 1 above)
Comment (Fil158) [reply news/warning.rst:11]: Ok, next time I'll do so

Comment (bobleesj) [reply news/warning.rst:11]: No please do now. These are user facing news.

File: src/diffpy/fourigui/fourigui.py (hunk 2 above)
Comment (Fil158) [reply src/diffpy/fourigui/fourigui.py:397]: Here is the worning, please see the [Numpy official documentation](https://numpy.org/doc/2.1/reference/generated/numpy.fft.fftn.html#numpy.fft.fftn)

image

To solve it I specify axes which is a list of integers with the same size of s
image

After the changes pytests run correctly and fourigui app works with test data.

Comment (sbillinge) [reply src/diffpy/fourigui/fourigui.py:397]: Thanks @Fil158 this is very helpful.  Next we need a quick discussion about what the shape of the arrays are and what th axes are and what our desired fft is.

Then finally, what code we think is best to accomplish that.

**Comment by cadenmyers13** [comment]
@sbillinge @Fil158 `fftholder` is (101, 101, 101) and there are three axes, (0, 1, 2). Fourigui allows you to explore reciprocal space in 3D. The user specifies an axis (0,1, or 2) then the slider in the gui (which goes from 0 to 100 in the test case) allows you to look at a specific slice, hence the shape (101, 101, 101). This seems to me to be on the right track. Here's an image of the gui

Screenshot 2024-11-07 at 4 34 58 PM

**Comment by cadenmyers13** [comment]
I don't know if my inline comments are showing to everyone... so I'm pasting them here. Sorry if you see duplicates.

@sbillinge Here's some info from the `np.fft.fftn()` documentation. If we specify `s` we must specify `axes` in numpy2. To make it more readable, maybe we can change it to `axes = list(range(fftholder.ndim))`?

Screenshot 2024-11-07 at 11 55 35 AM

**Comment by bobleesj** [comment]
@Fil158 you may close this issue if @cadenmyers13 is willing to do so.

**Pull request closed without merging**